server/attributes.c

```diff
@@ -83,7 +83,7 @@ void get_attr_change_set(
                     attr->owner, attr->data, i, string, sizeof(string));
                 if (strncmp(string, attr->last_values[i], MAX_RESULT_LENGTH))
                 {
-                    attr->update_index[i] = get_change_index();
+                    attr->update_index[i] = report_index + 1;
```

Comment:
I am not convinced by this.  This needs a bit more of a detailed argument for this change, and a review of how _all_ other change reporting is done, in particular how does this affect other uses of `get_change_index()`?

What bug or misbehaviour is this intended to fix?

Reply:
Without this change, sending `*CHANGES.ATTR?` twice would report the same polled attributes twice, you can reproduce it by running the tests.
Given that polled attributes are updated when the changes are requested, using `report_index  + 1` behaves as expected, reporting the polled attributes in the current request but not in the next one.

Reply:
yes indeed, it is unlike the other uses because the other uses happens asynchronously to getting the changes, while this use happens while getting the changes, Do you have some alternative way in mind which I don't see?

Reply:
> I'm still quite concerned. I don't remember how the change index filtering works, and I'm concerned that this is completely unlike all the other uses.

Sorry I failed to read between lines, here is how the change index work:
- A piece of information changes asynchronously and it gets a new index (atomically incrementing and fetching a global counter).
- The next time we fetch changes, we compare the piece of information counter value with the counter value from last fetch changes operation, additionally, we update the counter for next fetch to work properly. 

The reason why `report_index + 1` is needed in this case is because we want the comparison to be true for the current fetch changes operation but not for the next one.